File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program18.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return the count of digits in between 3 and 7.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Display
{
	private:
		int iNo;

	public:

		Display(int iNo)
		{
			this->iNo = iNo;
		}	

		int CountRange()
		{
			int iDigit = 0,iCnt=0;

			if(iNo<0)
			{
				iNo = -iNo;
			}

			while(iNo>0)
			{
				iDigit = iNo%10;
				if(iDigit>3 && iDigit<7)
				{
					iCnt++;
				}
				iNo = iNo/10;
			}	

			return iCnt;
		}

};

int main()
{
	int iValue=0,iRet=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	iRet = dObj->CountRange();

	cout<<"Range is: "<<iRet<<"\n";
	return 0;
}

/*
OUTPUT:

Enter Number: 4521
Range is: 2

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program18.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return the count of digits in between 3 and 7.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public int CountRange()
	{
                      	
		int iCnt=0;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			if(iDigit>3 && iDigit<7)
			{
				iCnt++;
			}
			iValue = iValue/10;
		}        
		return iCnt;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		iRet = obj.CountRange();
	
		System.out.println("Frequency is: "+iRet);
	}
}

/*
OUTPUT:

Enter number:
2395
Frequency is: 1

*/
